logging/logging/__init__.py

```diff
@@ -40,11 +39,17 @@ def isEnabledFor(self, level):
 
     def log(self, level, msg, *args):
         if level >= (self.level or _level):
-            msg = msg if not args else msg % args
-            record = "%s:%s: %s" % (self._level_str(level), self.name, msg)
-            if _stream is not None:
-                _stream.write(record)
-                _stream.write("\n")
+            record = LogRecord(
+                name=self.name,
```

Comment:
As argument order is known here, I'd suggest to pass positional args, for efficiency.